packages/preset-wind/src/rules/container.ts

```diff
@@ -10,34 +11,39 @@ export const container: Rule<Theme>[] = [
     (m, context) => {
       const { theme, variantHandlers } = context
 
-      let width = '100%'
-
       const themePadding = theme.container?.padding
       let padding: string | undefined
 
-      if (typeof themePadding === 'string')
+      if (isString(themePadding))
         padding = themePadding
       else
         padding = themePadding?.DEFAULT
 
+      const themeMaxWidth = theme.container?.maxWidth
+      let maxWidth: string | undefined
+
       for (const v of variantHandlers) {
         const query = v.handle?.({} as VariantHandlerContext, x => x)?.parent
-        if (typeof query === 'string') {
+        if (isString(query)) {
           const match = query.match(queryMatcher)?.[1]
           if (match) {
-            width = match
-
             const bp = resolveBreakpoints(context) ?? {}
             const matchBp = Object.keys(bp).find(key => bp[key] === match)
 
-            if (matchBp && typeof themePadding !== 'string')
+            if (!themeMaxWidth)
+              maxWidth = match
+            else if (matchBp)
+              maxWidth = themeMaxWidth?.[matchBp]
+
+            if (matchBp && !isString(themePadding))
               padding = themePadding?.[matchBp] ?? padding
           }
         }
       }
 
       const css: CSSObject = {
-        'max-width': width,
+        'width': '100%',
+        'max-width': maxWidth,
```

Comment:
```suggestion
        'max-width': maxWidth || '100%',
```

Can we do this and avoid adding a new attribute?

Reply:
The reason that was added so the content is always expand the container. If we remove the `width` then it would not get that effect.